client/src/pages/UserAdmin.js

```diff
@@ -0,0 +1,318 @@
+                    "x-customrequired-header": headerToSend
+                  }
+            });
+            const resJson = await res.json();
+            setUsers(resJson);
+
+        } catch(error) {
+            alert(`fetchUsers ${error}`);
+        }
+    }
+
+    // Fetch projects from db
+    async function fetchProjects() {
+        try {
+            const res = await fetch("/api/projects/", {
+                headers: {
+                    "x-customrequired-header": headerToSend
+                    }
+            });
+            const resJson = await res.json();
+            setProjects(resJson);
+
+        } catch(error) {
+            alert(`fetchProjects: ${error}`);
+        }
+    }
+
+    useEffect(() => {
+        fetchUsers();
+        fetchProjects();
+    }, []);
+
+    // Handle click on selected user in search results
+    const userClickHandler = usr => event => {
+        setUserToEdit(usr);
+    }
+
+
+    // Get search resuts to display on page
+    const emailResults = !searchTerm
+    ? []
+    : Object.values(users).filter 
+        (user => user.email.toLowerCase().startsWith(searchTerm.toLowerCase().trim()))
+        .sort((a,b) => a.email.localeCompare(b.email))
+        .map((u) => <div onClick={userClickHandler(u)}>{u.email + " ( " + u.name.firstName + " " + u.name.lastName + " )"}</div>)      
+    ;
+    
+    const nameResults = !searchTerm
+    ? []
+    : Object.values(users).filter 
+        (user => 
+            user.name.firstName.concat(' ', user.name.lastName).toLowerCase().startsWith(searchTerm.toLowerCase().trim()))
+            .sort((a,b) => a.name.firstName.concat(a.name.lastName).localeCompare(b.name.firstName.concat(b.name.lastName)))
+            .map((u) => <div onClick={userClickHandler(u)}>{u.name.firstName + " " + u.name.lastName + " ( " + u.email + " )"}</div>)
+    ;
+
+    // Filter active projects for dropdown
+    const activeProjects = Object.values(projects).filter (project => project.projectStatus === 'Active')
+    .map((p) => [p._id, p.name])
+    ;
+
+    // Handle change on input in search form
+    const handleChange = event => {
+        setSearchTerm(event.target.value);
+    };
+
+    // Updates user projects in db
+    const updateUserDb = async () => {
+
+        // // renaming variable so it matches db name
+
+        let managedProjects = userManagedProjects;
+
+        // // Update database
+        const url = `/api/users/${userToEdit._id}`;
+        const requestOptions = {
+            method: 'PATCH',
+            headers: {
+                "Content-Type": "application/json",
+                "x-customrequired-header": headerToSend
+            },
+            body: JSON.stringify({ managedProjects })
+        };
+
+        try {
+            const response = await fetch(url, requestOptions); 
+            const resJson = await response.json();
+            return resJson;
+        } catch (error) {
+            console.log(`update user error: `, error);
```

Comment:
Add alert with the following message: "Server doesn't respond. Please try again."

Reply:
fixed